src/background/services/keyAutoAdd.ts

```diff
@@ -82,13 +85,20 @@ export class KeyAutoAddService {
     return this.tab?.id;
   }
 
-  private async process(url: string, payload: BeginPayload, tabId?: TabId) {
+  private async process(
+    url: string,
+    payload: BeginPayload,
+    existingTabId?: TabId,
+  ) {
     const { resolve, reject, promise } = withResolvers();
+    let tab: Tabs.Tab;
+    try {
+      tab = await this.browser.tabs.get(existingTabId ?? -1);
+      await this.browser.tabs.update(tab.id, { url });
+    } catch {
+      tab = await this.browser.tabs.create({ url });
+    }
```

Comment:
Let's extract this into a helper function, `reuseOrCreateTab(browser, url, tabId?): Promise<Tab>` in `background/utils`.

Reply:
if extract, let's also use `reuseOrCreateTab` in `getInteractionInfo` now. what do you think?